File: 011Container-with-Most-Water/011.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxarea = 0
        l = 0
        r = len(height) - 1
        while l<r:
            maxarea = max(maxarea, min(height[l], height[r]) * (r - l))
            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return maxarea
        

File: 011Container-with-Most-Water/011.c
int maxArea(int* height, int heightSize) {
    int maxarea = 0, l = 0, r = heightSize - 1;
    while (l < r) {
        maxarea = max(maxarea, min(height[l], height[r]) * (r - l));
        if (height[l] < height[r])
                l++;
        else
                r--;
        }
    return maxarea;
}

int max(int a,int b){
    if(a>b)
        return a;
    else
        return b;
}

int min(int a,int b){
    if(a<b)
        return a;
    else
        return b;
}
